core/rint/test/TTabComTests.cxx

```diff
@@ -71,7 +71,10 @@ TEST(TTabComTests, CompleteTH1)
       // FIXME: See ROOT-10989
       " TH1DModel"
 #endif
-      " TH1Editor TH1F TH1I TH1L TH1S";
+#ifndef R__IS_MINIMAL
```

Comment:
```suggestion
#if not defined(ROOT_IS_MINIMAL)
```
I would prefer to not add a ROOT runtime preprocessor macro `R__IS_MINIMAL` that is also visible to the user if there is no strong reason for it. Because then we need to keep it in principle forever for backwards compatibility, locking ourselves in without a strong reason. Instead, one can add a
```cmake
if (minimal)
    target_compile_definitions(TTabComTests PRIVATE ROOT_IS_MINIMAL)
endif()
```
to the [CMakeLists.txt](https://github.com/root-project/root/blob/master/core/rint/test/CMakeLists.txt). The runtime `R__` flags only need to be defined if we need to do these conditionals also in the header files included by the users, but here it's just for a test.

Reply:
I don't understand how the `R__IS_MINIMAL` macro would sustainably address the issue you describe.

If it's useful for the user to check if ROOT has a certain feature, we should introduce `R__HAS_FEATURE` macros like we always did.

The `R__IS_MINIMAL` would be mute, because one can toggle any an feature on top of a minimal build. Why would you use it then?